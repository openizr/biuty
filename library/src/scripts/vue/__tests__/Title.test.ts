/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { mount } from '@vue/test-utils';
import UITitle from 'scripts/vue/Title.vue';

describe('vue/UITitle', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly - basic', () => {
    const wrapper = mount(UITitle, {
      propsData: { level: '1', modifiers: 'large' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - modifier different than level', () => {
    const wrapper = mount(UITitle, {
      propsData: { label: 'Test', level: '1', modifiers: '5 large' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with id', () => {
    const wrapper = mount(UITitle, {
      propsData: { label: 'Test', id: 'test' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with itemprop', () => {
    const wrapper = mount(UITitle, {
      propsData: { label: '*Test*', itemProp: 'name' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
