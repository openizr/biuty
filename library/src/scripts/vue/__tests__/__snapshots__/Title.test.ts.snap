// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue/UITitle renders correctly - basic 1`] = `"<h1 class=\\"ui-title ui-title--large--1\\"></h1>"`;

exports[`vue/UITitle renders correctly - modifier different than level 1`] = `"<h1 class=\\"ui-title ui-title--5--large\\">Test</h1>"`;

exports[`vue/UITitle renders correctly - with id 1`] = `"<h1 id=\\"test\\" class=\\"ui-title ui-title--1\\">Test</h1>"`;

exports[`vue/UITitle renders correctly - with itemprop 1`] = `"<h1 itemprop=\\"name\\" class=\\"ui-title ui-title--1\\"><span class=\\"ui-markdown--emphasis\\">Test</span></h1>"`;
